opamp-client/src/main/java/io/opentelemetry/opamp/client/internal/connectivity/http/RetryAfterParser.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opamp.client.internal.connectivity.http;
+
+import io.opentelemetry.opamp.client.internal.tools.SystemTime;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+public final class RetryAfterParser {
+  private final SystemTime systemTime;
+  public static final Pattern SECONDS_PATTERN = Pattern.compile("\\d+");
+  public static final Pattern DATE_PATTERN =
+      Pattern.compile(
+          "^([A-Za-z]{3}, [0-3][0-9] [A-Za-z]{3} [0-9]{4} [0-2][0-9]:[0-5][0-9]:[0-5][0-9] GMT)$");
```

Comment:
Only the `DATE_PATTERN` is surrounded with `^$` which is a bit strange as it is matched the same way as `SECONDS_PATTERN`. For consistency both regexes should either have or not have the line start/end anchors.